Sources/NostrSDK/Events/DirectMessageEvent.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  DirectMessageEvent.swift
+//  
+//
+//  Created by Joel Klabo on 8/10/23.
+//
+
+import Foundation
+
+/// An event that contains an encrypted message
+///
+/// > Note: [NIP-04 Specification](https://github.com/nostr-protocol/nips/blob/master/04.md)
+public final class DirectMessageEvent: NostrEvent, DirectMessageEncrypting {
+
+    public func decryptedContent(keypair: Keypair) throws -> String {
+        let recipient = tags.first { tag in
+            tag.name == .pubkey
+        }
+
+        guard let recipientPublicKeyHex = recipient?.value, let recipientPublicKey = PublicKey(hex: recipientPublicKeyHex) else {
+           fatalError()
```

Comment:
Since Nostr is an open protocol and anybody can do anything, including other clients, I'd prefer not to crash the client app when the data isn't as expected. Can we fail more gracefully here? I'd prefer either 1) throw a custom error, or 2) return nil.

Reply:
Oh yeah, meant to come back to that.